pkg/client/group_mgmt_client.go

```diff
@@ -113,12 +133,22 @@ func (client *GroupMgmtClient) Post(path string, payload interface{}, respHolder
 	if err != nil {
 		return nil, err
 	}
+
 	// IsSuccess method returns true if HTTP status `code >= 200 and <= 299` otherwise false.
 	if !response.IsSuccess() {
 		errResp, err := unwrapError(response.Body())
 		if err != nil {
 			return nil, err
 		}
+		// refresh session token
+		isSessionRefresh, err := client.refreshSessionToken(response.StatusCode(), errResp)
+		if err != nil {
+			return nil, err
```

Comment:
It might be unexpected for the caller to receive an error that did not come in response to their request.  An alternative is to return the original 401 error, but that doesn't seem better.

Will we advertise the automatic session management somewhere, like in the README?  That might be the place to mention this little quirk.

Reply:
May contain in the documentation.